routes/books.js

```diff
@@ -48,56 +91,29 @@ router.post('/', (req, res, next) => {
 })
 
 router.put('/:id', (req, res, next) => {
-  // Make sure that id is a number
-  if (!req.params.id || Number.isNaN(req.params.id)) {
-    throw new Error('Invalid ID')
-  }
-
-  knex('books')
-    .where('id', req.params.id)
-    .first()
-    .then(book => {
-      if (!book) {
-        return next()
-      }
-
-      return knex('books')
-        .update({
-          title: req.body.title,
-          author: req.body.author,
-          genre: req.body.genre,
-          description: req.body.description,
-          cover_url: req.body.coverUrl,
-        }, '*')
-        .where('id', req.params.id)
-    })
-    .then(books => {
-      if (!books) { return {message: 'nothing found for given id'} }
-      const newObj = {
-        id: books[0].id,
-        title: books[0].title,
-        author: books[0].author,
-        genre: books[0].genre,
-        description: books[0].description,
-        coverUrl: books[0].cover_url,
-      }
-      res.send(newObj)
-    })
-    .catch(err => {
-      next(err)
-    })
+  return Promise
+  .resolve(checkInput(req.params))
+  .then(getOne)
+  .then(book => {
+    if (!book || book.length === 0) {
+      return res.status(404).send({message: 'Not found'})
+    }
+    return req.body
+  })
+  .then(update)
+  .then(books => {
+    formatBook(books)
+    res.send(newObj)
+  })
+  .catch(err => {
+    next(err)
+  })
 })
 
 router.delete('/:id', (req, res, next) => {
   let book = null
-
-  if (!req.params.id || Number.isNaN(req.params.id)) {
-    throw new Error('Invalid ID')
-  }
-
-  knex('books')
-    .del()
-    .where('id', req.params.id)
+  let id = checkInput(req.params)
```

Comment:
**Skills Level-up:**  See if you can refactor `checkInput()` so it's similar to [how I used `validateId()` over in the solution branch](https://github.com/gSchool/g59-dry-refactor/blob/f6687478d3f2f4b039674fc9e76961f4199ffd3d/routes/books.js#L29-L33)

Note: all routes with `/:id` need the same validate technique.